modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -156,20 +219,54 @@ public void dispose()
         disposeSource();
     }
 
-    private void restartSource()
+    private class SourceRetryCallback implements RetryCallback
     {
-        try
+
+        private boolean firstTry = true;
```

Comment:
keeping this state seems dangerous since you're creating a new instance every time you go through ExtensionMessageSource#start():90